src/devices/ExplorerHat/Led.cs

```diff
@@ -76,7 +80,10 @@ protected virtual void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    Off();
+                    if (_shouldDispose)
```

Comment:
Looks like my last comment on here hasn't been addressed yet. For convenience, this was it:

```
This is not really disposing, this is only really turning the LED off. I realize you didn't introduce this, but now that you are touching this do you mind fixing that? Just dispose the controller after calling Off. Also, I believe Off should be called regardless in this case.
```